lib/checkstl.cpp

```diff
@@ -395,6 +469,74 @@ void CheckStl::iterators()
     }
 }
 
+bool CheckStl::compareIteratorAgainstDifferentContainer(const Token* operatorTok, const Token* containerTok, const unsigned int iteratorId, const std::map<unsigned int, const Token*>& iteratorScopeBeginInfo)
+{
+    if (!containerTok)
+        return false;
+
+    const Token *otherOperand = nullptr;
+    OperandPosition operandPosition = OperandPosition::None;
```

Comment:
Technically, it is redundant to initialize `operandPosition`.. and the enum constant `None` is not needed.

Reply:
@danmar Yes, you are right, I have made a fix.